Integration for final week

Quinn-5 opened the pull request.

strimper13 committed:
Vision Code + Network Table

updated vision code and network table put

strimper13 committed:
Add Grippy stuff

strimper13 committed:
Stream Contour

Streams COntours grando

strimper13 committed:
Vision Angles & Brightness v.1

Added Brightness features to allow for tape to be GREEN and begun playing with math for image angles on the contours

strimper13 committed:
Vision Angles v.2

Work in progress towards angle recognition

strimper13 committed:
Found Pairs of rectangles

First Search for the -15 degree contour and add to first part of pair,
#then search for -75 degree contour, put in pair, assure the -75 contour is to the right of the -15 degree contour,
#then assure the -75 contour is the closest contour to the -15,
#by searching for other -75 contours that are either closer or farther

strimper13 committed:
Determines Best Pair of Rectangles

strimper13 committed:
Fixed Network Tables for new code

Updates function to be through -1 to 1 again

strimper13 committed:
Attempted to fix pull request changes

lbrenn7 committed:
Initial Integration work

This mostly changes the robot maps. It also adds all of the subsystem components to the dashboard

lbrenn7 committed:
Completed Drive Command

Drive command works manually and encoders work, correct angles and distance can be measured

1000DollarBill committed:
Lift Integration

bot190 committed:
More lift integration

Update 3rd party libraries and use brake for controlling the lift

1000DollarBill committed:
Further integration.

-Corrections to Maflipulator logic
-Button mapping
-new "Crappy Lift" command for manual flipping

strimper13 committed:
Now Maflips for real

fixed Alex's code

strimper13 committed:
Prepped Camera for vision & prepped for vision testing

strimper13 committed:
Streams through RoboRIO just badly 

need to get real code on stream

strimper13 committed:
Got Vision Working to the point of align

msoucy committed:
Merge branch 'master' into vision

strimper13 committed:
Fixed Vision multiplier and Matt's comments round two

strimper13 committed:
Actually Fixed Matt's Comments

Quinn-5 committed:
Fixes to lift code

Add functions for isAtUpperLimit and isAtLowerLimit, delete empty flipOpp command

Quinn-5 committed:
reformat lift code and organize lift inputs

Quinn-5 committed:
Reformat and organize inputs of all java files

msoucy commented on .vscode/settings.json:
```diff
@@ -12,5 +12,6 @@
   },
-  "wpilib.online": true
+  "wpilib.online": true,
+  "python.pythonPath": "C:\\Users\\xtron\\AppData\\Local\\Programs\\Python\\Python37-32\\python.exe"
```
Remove this line

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -27,19 +27,21 @@
 
-    UsbCamera camera0;
-    UsbCamera camera1;
+    UsbCamera cameraBack;
```
Mark these four variables private

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -95,28 +102,45 @@ public void teleopInit() {
-            camera0.setBrightness(100);
-            camera1.setBrightness(100);
+            cameraBack.setExposureAuto();
+            ;
```
This statement does nothing. As much as I like doing nothing, the robot doesn't.

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -210,10 +198,11 @@ public Command goToHeight(Heights height) {
+                brake.set(Value.kReverse);
 
-                if ((currentHeight < height.get()) && upperLimit.get()) {
+                if ((currentHeight < height.get()) && !upperLimit.get()) {
```
This should also use the functions that you made

msoucy commented on src/main/java/frc/robot/subsystems/Maflipulator.java:
```diff
@@ -46,47 +47,56 @@ public Maflipulator(final RobotMap.MaflipulatorMap map) { // NOPMD
                 flipSpeed = 0;
             }
-            if (flipSpeed < 0 && anglePot.get() <= FRONT_LOWER_ANGLE) {
+            if (flipSpeed < 0 && anglePot.get() >= FRONT_LOWER_ANGLE) {
```
These should be `else if`

msoucy commented on src/main/java/frc/robot/subsystems/Maflipulator.java:
```diff
@@ -102,17 +112,56 @@ public Command Flip() {
 
```
Maybe look at using a `ConditionalCommand` here instead (optional).

Quinn-5 committed:
Separate lift up and down speeds, fix isAtUpper/LowerLimit functions

adds separate lift up and down speeds, finish implementing isAtLimit functions so they are now used everywhere they are needed

Quinn-5 committed:
Remove python path in settings

Quinn-5 committed:
define camera variables as private

Quinn-5 committed:
get rid of green squiggles in drive

The downOffDrop command group in Drive no longer has a 'resource leak', and doesn't show as a problem

DavidDesfosse committed:
Made everything Conditional Commands

Made go to scoring position, changed flip to a conditional command

msoucy reviewed (approved):
LGTM :shipit:
